plumbing/clock/clock.go

```diff
@@ -0,0 +1,29 @@
+package clock
+
+import (
+	"time"
+)
+
+// BlockClock defines an interface for fetching the block time.
+type BlockClock interface {
```

Comment:
This is the interface we can extend, adding a method like `EpochSeconds`. Miners will then call this method when adding a timestamp to the block.